src/main/java/org/apache/tools/bzip2/CBZip2InputStream.java

```diff
@@ -29,957 +29,918 @@
+        if (offs < 0) {
+            throw new IndexOutOfBoundsException("offs(" + offs + ") < 0.");
+        }
+        if (len < 0) {
+            throw new IndexOutOfBoundsException("len(" + len + ") < 0.");
+        }
+        if (offs + len > dest.length) {
+            throw new IndexOutOfBoundsException("offs(" + offs + ") + len(" + len + ") > dest.length(" + dest.length + ").");
+        }
+        if (this.in == null) {
+            throw new IOException("stream closed");
+        }
+
+        final int hi = offs + len;
+        int destOffs = offs;
+        for (int b; (destOffs < hi) && ((b = read0()) >= 0); ) {
+            dest[destOffs++] = (byte) b;
+        }
+
+        return (destOffs == offs) ? -1 : (destOffs - offs);
+    }
+
+    private int read0() throws IOException {
+        final int retChar = this.currentChar;
+
+        switch (this.currentState) {
+            case EOF:
+                return -1;
+
+            case START_BLOCK_STATE:
+                throw new IllegalStateException();
+
+            case RAND_PART_A_STATE:
+                throw new IllegalStateException();
+
+            case RAND_PART_B_STATE:
+                setupRandPartB();
+                break;
+
+            case RAND_PART_C_STATE:
+                setupRandPartC();
+                break;
+
+            case NO_RAND_PART_A_STATE:
+                throw new IllegalStateException();
+
+            case NO_RAND_PART_B_STATE:
+                setupNoRandPartB();
+                break;
+
+            case NO_RAND_PART_C_STATE:
+                setupNoRandPartC();
+                break;
+
+            default:
+                throw new IllegalStateException();
+        }
+
+        return retChar;
+    }
+
+    private void init() throws IOException {
+        if (null == in) {
+            throw new IOException("No InputStream");
+        }
+        if (in.available() == 0) {
+            throw new IOException("Empty InputStream");
+        }
+        int magic2 = this.in.read();
+        if (magic2 != 'h') {
+            throw new IOException("Stream is not BZip2 formatted: expected 'h'" + " as first byte but got '" + (char) magic2 + "'");
+        }
+
+        int blockSize = this.in.read();
+        if ((blockSize < '1') || (blockSize > '9')) {
+            throw new IOException("Stream is not BZip2 formatted: illegal " + "blocksize " + (char) blockSize);
+        }
+
+        this.blockSize100k = blockSize - '0';
+
+        initBlock();
+        setupBlock();
+    }
+
+    private void initBlock() throws IOException {
+        char magic0 = bsGetUByte();
+        char magic1 = bsGetUByte();
+        char magic2 = bsGetUByte();
+        char magic3 = bsGetUByte();
+        char magic4 = bsGetUByte();
+        char magic5 = bsGetUByte();
+
+        if (magic0 == 0x17 && magic1 == 0x72 && magic2 == 0x45 && magic3 == 0x38 && magic4 == 0x50 && magic5 == 0x90) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://titan.maxgamer.org:9000/coding_rules#rule_key=squid%3ACommentedOutCodeLine)